rotkehlchen/chain/ethereum/modules/kyber/decoder.py

```diff
@@ -51,14 +57,25 @@ def _maybe_update_events_legacy_contrats(
     destination_asset: CryptoAsset,
     spent_amount: FVal,
     return_amount: FVal,
+    msg_aggregator: MessagesAggregator,
 ) -> None:
     """
     Use the information from a trade transaction to modify the HistoryEvents from receive/send to
     trade if the conditions are correct.
     """
     in_event = out_event = None
     for event in decoded_events:
-        crypto_asset = event.asset.resolve_to_crypto_asset()
+        try:
+            crypto_asset = event.asset.resolve_to_crypto_asset()
+        except (UnknownAsset, WrongAssetType) as e:
+            log.error(f'Failed to decode airdrop event with asset {event.asset}. {str(e)}')
+            msg_aggregator.add_error(
```

Comment:
same here (since it's used in so many places consider moving the log to a superclass method and just repeat it or ... somehow try to abstract it. As you are basically copy pasting code everywhere here.

You can make the log much more specific if it's handled in a higher point. Take the name of the decoder and name of asset as parameters and just handle it higher.